rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -207,15 +207,15 @@ namespace rsx
 		{
 			if (rsx->in_begin_end)
 			{
-				rsx->append_array_element(arg & 0xFFFF);
-				rsx->append_array_element(arg >> 16);
+				rsx->append_array_element(se_storage<u16>::swap(arg & 0xFFFF));
+				rsx->append_array_element(se_storage<u16>::swap(arg >> 16));
```

Comment:
No, the value must be cast to a u32 and then swapped because that is the element size of the vector it gets pushed to.